Source/WebCore/rendering/RenderBlockFlow.cpp

```diff
@@ -2517,6 +2593,32 @@ LayoutUnit RenderBlockFlow::logicalRightOffsetForPositioningFloat(LayoutUnit log
     return adjustLogicalRightOffsetForLine(offset, applyTextIndent);
 }
 
+void RenderBlockFlow::trimMarginForFloat(FloatingObject& floatingObject, MarginTrimType marginTrimType, LayoutUnit trimAmount)
+{
+    switch (marginTrimType) {
+    case MarginTrimType::InlineStart: {
+        setLogicalWidthForFloat(floatingObject, logicalWidthForFloat(floatingObject) - floatingObject.renderer().marginStart(&style()));
+        floatingObject.renderer().setMarginStart(0_lu, &style());
```

Comment:
FloatingObject has a setMarginOffset() function. Do you know if these need to be in sync?

Reply:
Yes I think so. The margin offset for the float are being set after we trim them, `RenderBlockFlow::computeLogicalLocationForFloat` so we should be good for that case I believe. I also added an assert `ASSERT(!floatingObject.isPlaced() || marginTrimType == MarginTrimType::BlockEnd)` inside `RenderBlockFlow::trimMarginForFloat` so that this method does not accidentally get called after a float gets its position and gets out of sync with the margin offset

Reply:
👍